digdag-standards/src/main/java/io/digdag/standards/operator/td/TdWaitTableOperatorFactory.java

```diff
@@ -0,0 +1,148 @@
+import io.digdag.spi.OperatorFactory;
+import io.digdag.spi.TaskExecutionException;
+import io.digdag.spi.TaskRequest;
+import io.digdag.spi.TaskResult;
+import io.digdag.spi.TemplateEngine;
+import io.digdag.util.BaseOperator;
+import org.msgpack.core.MessageTypeCastException;
+import org.msgpack.value.ArrayValue;
+import org.msgpack.value.IntegerValue;
+import org.msgpack.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigInteger;
+import java.nio.file.Path;
+import java.time.Duration;
+
+import static io.digdag.standards.operator.td.TdOperatorFactory.joinJob;
+
+public class TdWaitTableOperatorFactory
+        extends AbstractWaitOperatorFactory
+        implements OperatorFactory
+{
+    private static Logger logger = LoggerFactory.getLogger(TdWaitTableOperatorFactory.class);
+
+    private final TemplateEngine templateEngine;
+
+    @Inject
+    public TdWaitTableOperatorFactory(TemplateEngine templateEngine, Config systemConfig)
+    {
+        super(systemConfig);
+        this.templateEngine = templateEngine;
+    }
+
+    public String getType()
+    {
+        return "td_wait_table";
+    }
+
+    @Override
+    public Operator newTaskExecutor(Path workspacePath, TaskRequest request)
+    {
+        return new TdWaitTableOperator(workspacePath, request);
+    }
+
+    private class TdWaitTableOperator
+            extends BaseOperator
+    {
+        public TdWaitTableOperator(Path workspacePath, TaskRequest request)
+        {
+            super(workspacePath, request);
+        }
+
+        @Override
+        public TaskResult runTask()
+        {
+            Config params = request.getConfig().mergeDefault(
+                    request.getConfig().getNestedOrGetEmpty("td"));
+
+            int pollInterval = getPollInterval(params);
+
+            TableParam table = params.get("_command", TableParam.class);
+
+            int rows = params.get("rows", int.class, 0);
+
+            boolean done = poll(params, table, rows);
+
+            if (done) {
+                return TaskResult.empty(request);
+            }
+            else {
+                throw TaskExecutionException.ofNextPolling(pollInterval, ConfigElement.copyOf(request.getLastStateParams()));
+            }
+        }
+    }
+
+    private boolean poll(Config params, TableParam table, int expectedRows)
+    {
+        String engine = params.get("engine", String.class, "presto");
+
+        String tableName;
+        switch (engine) {
+            case "presto":
+                tableName = TDOperator.escapePrestoTableName(table);
+                break;
+            case "hive":
+                tableName = TDOperator.escapeHiveTableName(table);
+                break;
+            default:
+                throw new ConfigException("Unknown 'engine:' option (available options are: hive and presto): " + engine);
+        }
+        String query = "select count(1) from " + tableName + " limit " + expectedRows;
```

Comment:
I think this count + limit is not working as expected. What you need is:

`select count(*) from (select * from "tableName" limit expectedRows) sub`